pkg/tsdb/azuremonitor/azure-log-analytics-datasource.go

```diff
@@ -240,24 +244,17 @@ func (e *AzureLogAnalyticsDatasource) createRequest(ctx context.Context, dsInfo
 
 	req.Header.Set("Content-Type", "application/json")
 
-	// find plugin
-	plugin := e.pluginManager.GetDataSource(dsInfo.Type)
-	if plugin == nil {
-		return nil, errors.New("unable to find datasource plugin Azure Monitor")
-	}
-
-	logAnalyticsRoute, routeName, err := e.getPluginRoute(plugin)
-	if err != nil {
-		return nil, err
-	}
-
-	pluginproxy.ApplyRoute(ctx, req, routeName, logAnalyticsRoute, dsInfo, e.cfg)
+	// TODO: Use backend authentication instead
```

Comment:
As far as I understood, this `proxy.ApplyRoute` could be removed with a change similar to https://github.com/grafana/azure-data-explorer-datasource/pull/231 so we don't need to manually build an object `&models.DataSource` here.

Reply:
Yes - this will be our next step in refactoring the AM backend. More info [here](https://github.com/grafana/grafana/issues/33809)